org/postgresql/xa/PGXAConnection.java

```diff
@@ -331,6 +332,12 @@ public int prepare(Xid xid) throws XAException {
         }
     }
 
+    private String currentDatabase() throws SQLException{
+    	DatabaseMetaData metaData = conn.getMetaData();
+    	Properties properties = Driver.parseURL(metaData.getURL(),new Properties());
```

Comment:
Can you please clarify how often this is supposed to be invoked?
If that happens for each transaction, it can have severe performance impact.

In general, it makes sense to cache that database name somewhere or fetch it from some existing variable.